samples/mpsl/timeslot/src/main.c

```diff
@@ -50,26 +54,54 @@ static mpsl_timeslot_request_t timeslot_request_normal = {
 
 static mpsl_timeslot_signal_return_param_t signal_callback_return_param;
 
-/* Message queue for printing the signal type from timeslot callback */
-K_MSGQ_DEFINE(callback_msgq, sizeof(uint32_t), 10, 4);
+/* Two ring buffers for printing the signal type with different priority from timeslot callback */
+RING_BUF_DECLARE(callback_high_priority_ring_buf, 10);
+RING_BUF_DECLARE(callback_low_priority_ring_buf, 10);
 
 /* Message queue for requesting MPSL API calls to non-preemptible thread */
 K_MSGQ_DEFINE(mpsl_api_msgq, sizeof(enum mpsl_timeslot_call), 10, 4);
 
-static void error(void)
+ISR_DIRECT_DECLARE(swi1_isr)
 {
-	printk("ERROR!\n");
-	while (true) {
-		/* Spin for ever */
-		k_sleep(K_MSEC(1000));
-	}
+	uint8_t signal_type = 0;
+
+	if (ring_buf_get(&callback_high_priority_ring_buf, &signal_type, 1) == 1) {
+			switch (signal_type) {
+			case MPSL_TIMESLOT_SIGNAL_START:
+				LOG_INF("Callback: Timeslot start\n");
+				break;
+			case MPSL_TIMESLOT_SIGNAL_TIMER0:
+				LOG_INF("Callback: Timer0 signal\n");
+				break;
+			default:
+				LOG_INF("Callback: Other signal: %d\n", signal_type);
+				break;
+			}
+		}
+
+	if (ring_buf_get(&callback_low_priority_ring_buf, &signal_type, 1) == 1) {
+			switch (signal_type) {
+			case MPSL_TIMESLOT_SIGNAL_SESSION_IDLE:
+				LOG_INF("Callback: Session idle\n");
+				break;
+			case MPSL_TIMESLOT_SIGNAL_SESSION_CLOSED:
+				LOG_INF("Callback: Session closed\n");
+				break;
+			default:
+				LOG_INF("Callback: Other signal: %d\n", signal_type);
+				break;
+			}
+		}
```

Comment:
Need to handle the case that there are multiple entries in the ring buffer when the SWI is triggered. Therefore, we need a loop here